### PR TITLE
feat: adding allocated judge email (FPLA-1746)

### PR DESCRIPTION
**Pull request by toireasam**
**Before creating a pull request make sure that:**

- [ ] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NotifyAllocatedJudgeEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    private final NotificationService notificationService;
+    private final AllocatedJudgeContentProvider allocatedJudgeContentProvider;
+    private final ObjectMapper mapper;
+
+    @EventListener
+    public void notifyAllocatedJudge(NotifyAllocatedJudgeEvent event) {
+        EventData eventData = new EventData(event);
+        CaseData caseData = mapper.convertValue(eventData.getCaseDetails().getData(), CaseData.class);
+
+        Map<String, Object> parameters = allocatedJudgeContentProvider
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NotifyAllocatedJudgeEventHandler.java:30]: Could we use custom objects for this instead of a Map, it will be easier to maintain and review

File: service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationEventHandlerTestData.java [hunk 2]
```diff
@@ -57,6 +58,15 @@ public static CallbackRequest appendSendToCtscOnCallback() {
             .build();
     }
 
+    public static Map<String, Object> getExpectedAllocatedJudgeNotificationParameters() {
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationEventHandlerTestData.java:61]: I would keep this in the `NotifyAllocatedJudgeEventHandlerTest` unless you think that other tests are going to be able to make use of it

File: service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NotifyAllocatedJudgeEventHandler.java (hunk 1 above)
Comment (toireasam) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NotifyAllocatedJudgeEventHandler.java:30]: cool will change :)

File: service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationEventHandlerTestData.java (hunk 2 above)
Comment (toireasam) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationEventHandlerTestData.java:61]: will do :)

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AllocatedJudgeContentProvider.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import uk.gov.hmcts.reform.fpl.model.notify.AllocatedJudgeTemplate;
+import uk.gov.hmcts.reform.fpl.service.email.content.base.AbstractEmailContentProvider;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class AllocatedJudgeContentProvider extends AbstractEmailContentProvider {
+
+    private final ObjectMapper mapper;
+
+    public AllocatedJudgeTemplate buildAllocatedJudgeNotificationParameters(CaseDetails caseDetails) {
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AllocatedJudgeContentProvider.java:18]: I feel like this name is overly verbose, from the class name we know that we are going to be using functions that are related to allocated judge so I would probs just call this something like buildNotificationParameters, I feel like this provides all the info necessary without being too wordy

**Comment by lewisbirks** [comment]
Code smell seems to be the public modifier on
```java
public class NotifyAllocatedJudgeEventHandlerTest
```
it can probably just be package private (no modifier) if you want to resolve the code smell.
Also the controller isn't tested, I guess there is very little to actually test in there so I don't know what would actually be a good test there. I guess we could verify that upon calling the method that a notification is sent but that is all I can think of.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/notify/AllocatedJudgeTemplate.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.reform.fpl.model.notify;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
```
Comment (Jamesnelson117) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/notify/AllocatedJudgeTemplate.java:7]: Don't think setter annotation is needed since fields are not final.

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/notify/AllocatedJudgeTemplate.java:7]: Setter is needed, if the fields were final then the setter wouldn't be needed

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AllocatedJudgeContentProvider.java (hunk 3 above)
Comment (toireasam) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AllocatedJudgeContentProvider.java:18]: cool makes sense :)

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AllocatedJudgeControllerSubmittedTest.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.ALLOCATED_JUDGE_TEMPLATE;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.callbackRequest;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(AllocatedJudgeController.class)
+@OverrideAutoConfiguration(enabled = true)
+public class AllocatedJudgeControllerSubmittedTest extends AbstractControllerTest {
+    private static final String allocatedJudgeEmail = "test@test.com";
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AllocatedJudgeControllerSubmittedTest.java:20]: constants should be uppercase

Comment (toireasam) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AllocatedJudgeControllerSubmittedTest.java]: fixed :) checkstyle complained initially. weird.
